### PR TITLE
stlog: Adapt and cover

### PR DESCRIPTION
**Pull request by ChriMarMe**
stlog/StandardLogger:
* Allow to pass writer into standard logger for testability
* Rewrite tests to increase coverage of standard logger

Signed-off-by: Christopher Meis <christopher.meis@9elements.com>

### CODE REVIEW

File: stlog/log_test.go [hunk 1]
```diff
@@ -4,53 +4,113 @@
...
+			case DebugLevel:
+				l.debug(tt.format, tt.input)
+			default:
+				fmt.Fprintf(&buf, "%s\n", "Invalid LogLevel")
+			}
+			got := buf.String()
+			if got != tt.want {
+				t.Errorf("Test: %s failed.\nGot : %sWant: %s", tt.name, got, tt.want)
+			}
+		})
```
Comment (jensdrenhaus) [inline stlog/log_test.go:76]: Well done! 
Maybe put this in `std_test.go`.

File: stlog/log_test.go [hunk 2]
```diff
@@ -4,53 +4,113 @@
...
+			default:
+				SetLevel(tt.level)
+			}
+		})
 
-	t.Logf("%+v", stl)
-	Debug("hello")
-	Error("fooo %d", 7)
-	Info("This %s is a %d", "bar", 7)
+	}
```
Comment (jensdrenhaus) [inline stlog/log_test.go:94]: Where is the test condition?
A meaningful thing would be to test if calls to `Error()`, `Warn()`, `Info()`, `Debug()` only produce output when the log level is set accordingly.

File: stlog/log_test.go [hunk 3]
```diff
@@ -4,53 +4,113 @@
...
+			input: "LogLevel 0",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			SetOutput(StdError)
+		})
+		t.Run(tt.name, func(t *testing.T) {
+			SetOutput(KernelSyslog)
+		})
+	}
```
Comment (jensdrenhaus) [inline stlog/log_test.go:115]: What is tested here?

**Review by jensdrenhaus — CHANGES_REQUESTED** [review]

File: stlog/log_test.go (hunk 3 above)
Comment (ChriMarMe) [reply stlog/log_test.go]: Only coverage

**Comment by jensdrenhaus** [comment]
@ChriMarMe take a look.
Do you have an idea of some smarter/shorter/more elegant way of testing log levels in `TestStandardLoggerLevel()` in std_test.go`?

**Comment by ChriMarMe** [comment]
@jensdrenhaus LGTM. Can't think of a different, more elegant way tbh.

**Review by jensdrenhaus — APPROVED** [review]

**Pull request merged**